File: pset6/cash.py
from cs50 import get_float

input = None
change = 0
cents = None
quarters = 0
nickels = 0
dimes = 0
pennies = 0


def CalcQuarters(Cents):
    Quarters = 0
    while Cents >= 0.25:
        Quarters += 1
        Cents -= 0.25
    return Quarters


def CalcNickels(Cents):
    Nickels = 0
    while Cents >= 0.05:
        Nickels += 1
        Cents -= 0.05
    return Nickels


def CalcDimes(Cents):
    Dimes = 0
    while Cents >= 0.10:
        Dimes += 1
        Cents -= 0.10
    return Dimes


def CalcPennies(Cents):
    Penny = 0
    while Cents >= 0.01:
        Penny += 1
        Cents -= 0.01
    return Penny


input = get_float("Change: ")

while input < 0:
    input = get_float("Change: ")

cents = input

quarters = CalcQuarters(cents)

change += quarters

cents -= quarters * 0.25

cents = round(cents, 2)

if (quarters == 0 or cents > 0):
    dimes = CalcDimes(cents)
    change += dimes
    cents -= dimes * 0.10
    cents = round(cents, 2)

if (dimes == 0 or cents > 0):
    nickels = CalcNickels(cents)
    change += nickels
    cents -= nickels * 0.05
    cents = round(cents, 2)

if (nickels == 0 or cents > 0):
    pennies = CalcPennies(cents)
    change += pennies
    cents -= pennies * 0.01
    cents = round(cents, 2)

print(change)

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    int CalcQuarters(int cents);
    int CalcDimes(int cents);
    int CalcNickels(int cents);
    int CalcPennies(int cents);
    int input;
    int change = 0;
    int cents;
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;
    // predefines everything so when we define the amounts they are on point

    do
    {
        input = get_int("Cents: ");
        // get cents
    }
    while (input < 0);
    // check to make sure value is positve

    cents = input;
    // sets cents to input so it doesn't directly affect input just to avoid errors
    quarters = CalcQuarters(cents);
    // sets variable quarters to how many quarter you need for currents cents
    change += quarters;
    // adds it to final total
    cents -= quarters * 25;
    // subtracts from cents so we know when we hit our input amount
    // this does what we do with quarters but with different values, makes sure there ethier is
    // still cents left or the value was too  low for the last amt
    if (quarters == 0 || cents > 0)
    {
        dimes = CalcDimes(cents);
        change += dimes;
        cents -= dimes * 10;
    }
    if (dimes == 0 || cents > 0)
    {
        nickels = CalcNickels(cents);
        change += nickels;
        cents -= nickels * 5;
    }
   if (nickels == 0 || cents > 0)
    {

        pennies = CalcPennies(cents);
        change += pennies;
        cents -= pennies * 1;
    }

    // prints total
    printf("%i\n", change);
}

// for every time we are able to subtract from our total by the value of the coin it returns that
// amt, these all repeat except for the coin val
int CalcQuarters(int cents)
{
    int Quarters = 0;
    while (cents >= 25)
    {

        cents -= 25;
        Quarters++;
    }

    return Quarters;
}
int CalcDimes(int cents)
{

    int Dimes = 0;
    while (cents >= 10)
    {

        cents -= 10;
        Dimes++;
    }

    return Dimes;
}
int CalcNickels(int cents)
{

    int Nickels = 0;
    while (cents >= 5)
    {

        cents -= 5;
        Nickels++;
    }

    return Nickels;
}
int CalcPennies(int cents)
{

    int Pennies = 0;
    while (cents >= 1)
    {

        cents -= 1;
        Pennies++;
    }

    return Pennies;
}
